web/yo/app/scripts/controllers/gene.js

```diff
@@ -3009,6 +3035,184 @@ angular.module('oncokbApp')
+        $scope.addTherapyError = false;
+        $scope.drugList = data.drugList;
+        var therapyUuid = new Array();
+        if (data.modifymode === true) {
+            $scope.therapy = new Array();
+            initTherapy();
+        } else {
+            $scope.therapy = [[]];
+            therapyUuid = [[]];
+        }
+
+        function initTherapy() {
+            var newTherapy = [];
+            newTherapy = data.treatmentRef.name.split(",");
+            for (var i = 0; i < newTherapy.length; i++) {
+                newTherapy[i] = newTherapy[i].trim();
+                newTherapy[i] = newTherapy[i].split(" + ");
+            }
+            ;
+            firebase.database().ref('Drugs').once('value').then(function (snapshot) {
+                var druglist = new Array();
+                druglist = snapshot.val();
+                for (var i = 0; i < newTherapy.length; i++) {
+                    $scope.therapy.push({});
+                    therapyUuid.push([]);
+                    var tem = new Array();
+                    var temuuid = new Array();
+                    for (var j = 0, arr = []; j < newTherapy[i].length; j++) {
+                        tem.push(druglist[newTherapy[i][j]]);
+                        temuuid.push(druglist[newTherapy[i][j]].uuid);
+                    }
+                    $scope.therapy[i] = {tags: tem};
+                    therapyUuid[i] = temuuid;
+                }
+            }, function (error) {
+                console.error((error));
+            });
+            $scope.therapy.push({});
+            therapyUuid.push([]);
+        }
+
+        $scope.loadDrugs = function ($query) {
+            return firebase.database().ref('Drugs').once('value').then(function (snapshot) {
+                var drugs = new Array();
+                $scope.addTherapyError = false;
+                snapshot.forEach(function (drug) {
+                    var drugVal = drug.val();
+                    drugs.push(drugVal);
+                })
+                return drugs.filter(function (drug) {
+                    if (!drug.synonyms)
+                        return drug.drugName.toLowerCase().indexOf($query.toLowerCase()) != -1
+                    else return (drug.drugName.toLowerCase().indexOf($query.toLowerCase()) != -1) || (drug.synonyms.join(',').toLowerCase().indexOf($query.toLowerCase()) != -1);
+                })
+            });
+        }
+
+        $scope.addDruginElement = function (tag, index) {
+            therapyUuid[index].push(tag.uuid);
+            addTherapy(index);
+        }
+
+        $scope.removeDruginElement = function (tag, index) {
+            var id = therapyUuid[index].indexOf(tag.uuid);
+            if (id > -1) {
+                therapyUuid[index].splice(id, 1);
+            }
+            addTherapy(index);
+        }
+
+
+        function addTherapy(index) {
```

Comment:
At start, there is only one empty line. When the user have chosen a tag for the first line, the second empty line will be added. Being called in addDruginElement is for adding an empty line automatically while users adding a tag.
What's more, if there were two lines now, and the user clicked the second delete button to delete the second line, and then she also wanted to delete a drug instead of adding a new drug in the first line, calling this function in removeDruginElement would help add an empty line. The user may feel more comfortable.
In a word, calling this function in both help keep same, thus any changes in last line will lead to a new line.